astropy/wcs/wcs.py

```diff
@@ -3045,21 +3045,21 @@ def has_celestial(self):
     def pixel_scale_matrix(self):
 
         try:
-            cdelt = np.matrix(np.diag(self.wcs.get_cdelt()))
-            pc = np.matrix(self.wcs.get_pc())
+            cdelt = np.diag(self.wcs.get_cdelt())
+            pc = self.wcs.get_pc()
         except InconsistentAxisTypesError:
             try:
                 # for non-celestial axes, get_cdelt doesn't work
-                cdelt = np.matrix(self.wcs.cd) * np.matrix(np.diag(self.wcs.cdelt))
+                cdelt = np.dot(self.wcs.cd, np.diag(self.wcs.cdelt))
```

Comment:
It might not be.  I don't think this is the right PR to change that, since this is a simple refactoring for a future deprecation warning.  We might test against a non-compliant header that mixes CDELT and CD to see if this is needed?

Reply:
yes, please have that in a separate PR to keep this current one a simple backport to LTS.